mdgen/src/main/java/uk/gov/ida/mdgen/MetadataGenerator.java

```diff
@@ -189,20 +221,45 @@ private void sign(EntityDescriptor entityDescriptor) throws SecurityException, M
         SignatureValidator.validate(entityDescriptor.getSignature(), metadataSigningCredential);
     }
 
-    private void updateSsoDescriptor(EntityDescriptor entityDescriptor) throws SecurityException {
+    private void updateSsoDescriptors(EntityDescriptor entityDescriptor) throws Exception {
+        addSamlSigningKeyDescriptor(getSsoDescriptor(entityDescriptor));
+        addSamlEncryptionDescriptor(getSsoDescriptor(entityDescriptor));
+    }
+
+    private SSODescriptor getSsoDescriptor(EntityDescriptor entityDescriptor) {
+        SSODescriptor spSso = null;
         switch (nodeType) {
             case connector:
-                SPSSODescriptor spSso = entityDescriptor.getSPSSODescriptor(SAMLConstants.SAML20P_NS);
-                spSso.getKeyDescriptors().add(buildKeyDescriptor(UsageType.SIGNING, samlSigningCredential));
-                // TODO use a separate encryption cert
-                spSso.getKeyDescriptors().add(buildKeyDescriptor(UsageType.ENCRYPTION, samlSigningCredential));
+                spSso = entityDescriptor.getSPSSODescriptor(SAMLConstants.SAML20P_NS);
                 break;
-
             case proxy:
-                IDPSSODescriptor idpSso = entityDescriptor.getIDPSSODescriptor(SAMLConstants.SAML20P_NS);
-                idpSso.getKeyDescriptors().add(buildKeyDescriptor(UsageType.SIGNING, samlSigningCredential));
+                spSso = entityDescriptor.getIDPSSODescriptor(SAMLConstants.SAML20P_NS);
                 break;
         }
+        return spSso;
+    }
+
+    private void addSamlSigningKeyDescriptor(SSODescriptor spSso) throws Exception {
+        if ( SamlSigningCert.metadataControllerSuppliedSamlSigningCert != null ) {
```

Comment:
```suggestion
        if (SamlSigningCert.metadataControllerSuppliedSamlSigningCert != null) {
```
Spaces inside parenthesis 😬